File: GXpert/GXpert.Web/Modules/Exams/Exam/ExamWizardForm.cs

using Serenity.ComponentModel;
using System;

namespace GXpert.Exams.Forms;

[FormScript("Exams.ExamWizardForm")]
//[BasedOnRow(typeof(ExamRow), CheckNames = true)]
public class ExamWizardForm
{
    [Tab("Exam Details")]
    public string Title { get; set; }
    [HalfWidth]
    public short EExamType { get; set; }
    [HalfWidth]
    public short EExamState { get; set; }
    [HalfWidth]
    public string QuestionPaperPath { get; set; }
    [HalfWidth]
    public string ModelAnswerPaperPath { get; set; }
    [HalfWidth]
    public int ExamDuration { get; set; }
    [HalfWidth]
    public float MaxMarks { get; set; }
    [HalfWidth]
    public float NegativeMarks { get; set; }
 
    [HalfWidth]
    public short EResultType { get; set; }
  
    [HtmlContentEditor(Rows = 4), CssClass("fact")]
    public string Instructions { get; set; }
    [HalfWidth]
    public string SearchTags { get; set; }
    [Tab("Add Exam Sections")]
    [LabelWidth("0")]
    [ExamSectionEditor]
    public List<ExamSectionRow> ExamSection { get; set; }
    [Tab("Add Exam Questions")]
    [/*ExamQuestionsEditor*/ LabelWidth("0")]
    public List<ExamQuestionRow> ExamQuestions { get; set; }
}

File: GXpert/GXpert.Web/Modules/ServerTypes/Exams/ExamWizardForm.ts
import { ExamSectionEditor } from "@/Exams/ExamSection/ExamSectionEditor";
import { StringEditor, IntegerEditor, DecimalEditor, HtmlContentEditor, PrefixedContext, initFormType } from "@serenity-is/corelib";

export interface ExamWizardForm {
    Title: StringEditor;
    EExamType: IntegerEditor;
    EExamState: IntegerEditor;
    QuestionPaperPath: StringEditor;
    ModelAnswerPaperPath: StringEditor;
    ExamDuration: IntegerEditor;
    MaxMarks: DecimalEditor;
    NegativeMarks: DecimalEditor;
    EResultType: IntegerEditor;
    Instructions: HtmlContentEditor;
    SearchTags: StringEditor;
    ExamSection: ExamSectionEditor;
    ExamQuestions: StringEditor;
}

export class ExamWizardForm extends PrefixedContext {
    static readonly formKey = 'Exams.ExamWizardForm';
    private static init: boolean;

    constructor(prefix: string) {
        super(prefix);

        if (!ExamWizardForm.init)  {
            ExamWizardForm.init = true;

            var w0 = StringEditor;
            var w1 = IntegerEditor;
            var w2 = DecimalEditor;
            var w3 = HtmlContentEditor;
            var w4 = ExamSectionEditor;

            initFormType(ExamWizardForm, [
                'Title', w0,
                'EExamType', w1,
                'EExamState', w1,
                'QuestionPaperPath', w0,
                'ModelAnswerPaperPath', w0,
                'ExamDuration', w1,
                'MaxMarks', w2,
                'NegativeMarks', w2,
                'EResultType', w1,
                'Instructions', w3,
                'SearchTags', w0,
                'ExamSection', w4,
                'ExamQuestions', w0
            ]);
        }
    }
}